### DISCUSSION TITLE
CLBLAST: Resetting test?

### DISCUSSION THREAD

**Discussion by niansa** [ggml-org/llama.cpp#1350] [Q&A] [answered]
Hey!

When the OpenCL test was ran, I aborted it, and now it doesn't use OpenCL at all, and I have no idea how to make it retry.
Any ideas?

Thanks
niansa

**Comment by SlyEcho** [comment]
What is the OpenCL test?

I added the CLBlast version to the Windows releases, is there a problem you're having with it? I just tested and it seems to work for my integrated graphics on Windows 10, at least.

If you're talking about CMake checking for libraries, you can run it again giving the build directory and `-DLLAMA_CLBLAST=ON` and perhaps also `-DCLBlast_DIR=path/to/clblast/lib/cmake/CLBlast` if it is not installed where CMake expects to find it.

**Reply by niansa** [reply]
I am talking about that test that is performed on first run, on Linux at least.

**Reply by SlyEcho** [reply]
There is no test.

Are you talking about llama.cpp or some other program that is built on it?

**Reply by niansa** [reply]
There definitely is, in clblast, appearently. I have no idea what exactly it said or whatever, but it said something like "if there is an error message, it does not work", Idk.
I think I'll dig deeper into clblast itself.

**Reply by SlyEcho** [reply, marked as answer]
I think I understand now. This is in [ggml-opencl.c](https://github.com/ggerganov/llama.cpp/blob/master/ggml-opencl.c#L203).

The first run means first time in the program because it only needs to happen one time. Maybe a bit misleading.

There is no way it won't be run, there is no option to use OpenCL optionally if it is compiled into the program.

If it doesn't show, CLBlast is not compiled in.

If you use make then it is possible that some objects are compiled with different flags, do a `make clean` before or always use the `-B` flag.

**Reply by niansa** [reply]
thanks! Make clean did the trick :-)
